docs/recipes/index.txt

```diff
@@ -0,0 +1,99 @@
+.. _recipes:
+
+========================
+Recipes for common tasks
+========================
+
+How do I check out files with Dulwich?
+======================================
+
+The answer depends on the exact meaning of "check out" that is
+intended. There are several common goals it could be describing, and
+correspodingly several options to achieve them.
+
+Make sure a working tree on disk matches a particular commit (like ``git checkout``)
+------------------------------------------------------------------------------------
+
+:py:func:`dulwich.porcelain.checkout` is a very high-level function
+that operates on the working tree and behaves very similar to the
+``git checkout`` command. It packages a lot of functionality into a
+single command, just as Git's porcelain does, which is useful when
+matching Git's CLI is the goal, but might be less desireable for
+programmatic access to a repository's contents.
+
+Retrieve a single file's contents at a particular commit
```

Comment:
This is incomplete, since unlike checkout you'll be ignoring things like filters.